### PR TITLE
dxSpeedDialAction - Fix incorrect onClick handler is called after toggling visibility (T933671)

### PR DESCRIPTION
**Pull request by groshenkovamarina**

### CODE REVIEW

File: js/ui/speed_dial_action/speed_dial_item.js [hunk 1]
```diff
@@ -107,11 +107,11 @@ class SpeedDialItem extends Overlay {
     _getVisibleActions(actions) {
         const currentActions = actions || this.option('actions');
```
Comment (babich-a) [inline js/ui/speed_dial_action/speed_dial_item.js]:
```suggestion
        const currentActions = actions || this.option('actions')  || [];
```

File: js/ui/speed_dial_action/speed_dial_item.js [hunk 2]
```diff
@@ -107,11 +107,11 @@ class SpeedDialItem extends Overlay {
...
 
-        return currentActions.filter((action) => action.option('visible'));
+        return currentActions ? currentActions.filter((action) => action.option('visible')) : [];
```
Comment (babich-a) [inline js/ui/speed_dial_action/speed_dial_item.js]:
```suggestion
        return currentActions.filter((action) => action.option('visible'));;
```

File: js/ui/speed_dial_action/speed_dial_item.js [hunk 3]
```diff
@@ -107,11 +107,11 @@ class SpeedDialItem extends Overlay {
...
     }
 
     _getActionComponent() {
-        return this.option('actionComponent') || this._getVisibleActions()[0] || this.option('actions')[0];
+        return this._getVisibleActions().length === 1 ? this._getVisibleActions()[0] : this.option('actionComponent') || this.option('actions')[0];
```
Comment (babich-a) [inline js/ui/speed_dial_action/speed_dial_item.js]: I suggest writing some parts of this expression to the variable or use `if` operator.

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 4]
```diff
@@ -579,14 +579,19 @@ QUnit.module('add visible option', {
     });
 
     QUnit.test('check multiple value changes', function(assert) {
+        const firstClickStub = sinon.stub();
```
Comment (babich-a) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: It is better to make a separate test for this scenario and mention bug id in its header. Shorts tests are better (https://devexpress.sharepoint.com/:w:/s/devextreme/ESLCAi7OVW5LtMr9C7Fwp9MB_XSSpz88U9zWhokQcrhTng?e=tSETIV)

**Review by babich-a — CHANGES_REQUESTED** [review]

File: js/ui/speed_dial_action/speed_dial_item.js (hunk 3 above)
Comment (groshenkovamarina) [reply js/ui/speed_dial_action/speed_dial_item.js]: Done

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js (hunk 4 above)
Comment (groshenkovamarina) [reply testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Done

**Review by babich-a — APPROVED** [review]

**Pull request merged**